### DISCUSSION TITLE
Cached function in layout and page runs twice

### DISCUSSION THREAD

**Discussion by ghost** [vercel/next.js#69119] [Help]
### Link to the code that reproduces this issue

https://codesandbox.io/p/devbox/sparkling-water-jfcr4f

### To Reproduce

1. Open the sandbox above
2. Clear the terminal
3. Check the logs
4. Refresh the embedded browser
5. `getItem` will be logged twice

### Current vs. Expected behavior

Docs [here](https://nextjs.org/docs/app/building-your-application/data-fetching/fetching#orms-and-database-clients) state that:

> You can use [React cache](https://react.dev/reference/react/cache) to memoize data requests during a React render pass. For example, although the getItem function is called in both layout and page, only one query will be made to the database:

But on the Sandbox above, you can see in the logs that it gets called twice.

Tested in both Vercel deployment and locally.

### Provide environment information

```bash
Operating System:
  Platform: linux
  Arch: x64
  Version: #1 SMP PREEMPT_DYNAMIC Sun Aug  6 20:05:33 UTC 2023
Binaries:
  Node: 20.12.0
  npm: 10.5.0
  Yarn: 1.22.19
  pnpm: 8.15.6
Relevant Packages:
  next: 14.1.0
  eslint-config-next: 14.1.0
  react: 18.2.0
  react-dom: 18.2.0
  typescript: 5.3.3
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

Performance

### Which stage(s) are affected? (Select all that apply)

Vercel (Deployed)

**Comment by samcx** [comment]
Can you explain why you've implemented your `getItem` function the way you have? It looks you are returning a function in `cache()`, and then currying that (e.g., `cache(...)(id)`), which seems incorrect.

v0 was able to answer this too! → https://v0.dev/chat/CyEPxUC

**Comment by ghost** [comment]
@samcx this was a mistake, here I fixed it https://codesandbox.io/p/devbox/sparkling-water-jfcr4f. Same behavior as the original issue.

**Reply by samcx** [reply]
@kostas-pblworks I see the issue clearly now!

To clarify, `unstable_cache` ≠ [React Cache](https://react.dev/reference/react/cache).

We can indeed confirm that `unstable_cache` does not de-dupe—this can be added to fix this! I have notified the team.
